spec/javascripts/details/views/notification_bar_spec.js.coffee

```diff
@@ -13,30 +14,119 @@ describe 'Notification', ->
+      notification = new Notifications(getNotificationContent())  
+      el = $('
+        <meta name="flash-message" type="success" content="Notification text here">  
+      ')
+      notification._prepareOnLoadNotifications(el)
+      expect(notification.el.find('ul li').length).to.eql(1) 
+      expect(notification.el.find('ul li').is(':animated')).to.eql(false)
+      notification.el.find('ul li .close').trigger('click')
+      expect(notification.el.find('ul li').is(':animated')).to.eql(true)
   
   describe '#_prepare', ->
     it "prepare a notification", ->
-    it "handle a notification without a title", ->
+      notification = new Notifications(getNotificationContent())  
+      expect(notification.el.find('ul li').length).to.eql(0)
+      notice = notification._prepare({type: undefined, content: "", title: ""})
+      expect($(notice).hasClass('notice-type-primary')).to.eql(true)
+      expect($(notice).attr('data-timeout')).to.eql("5000")
+    it "handle a notification with alert", ->
+      notification = new Notifications(getNotificationContent())  
+      expect(notification.el.find('ul li').length).to.eql(0)
+      notice = notification._prepare({type: "alert", content: "", title: ""})
+      expect($(notice).attr('data-timeout')).to.not.eql("5000")
+      expect($(notice).attr('data-timeout')).to.eql("none")
     it "handle a notification without a content", ->
+      notification = new Notifications(getNotificationContent())  
+      expect(notification.el.find('ul li').length).to.eql(0)
+      notice = notification._prepare({type: "alert", content: "", title: ""})
+      expect($(notice).length).to.eql(1)
+      expect($(notice).find('.notification-content').text()).to.eql("")
     it "handle a notification without a type", ->
+      notification = new Notifications(getNotificationContent())  
+      expect(notification.el.find('ul li').length).to.eql(0)
+      notice = notification._prepare({type: "alert", content: "", title: ""})
+      expect($(notice).length).to.eql(1)
+      expect($(notice).find('.notification-title').text()).to.eql("")
     it "handle a notification with a timeout", ->
+      notification = new Notifications(getNotificationContent())  
+      expect(notification.el.find('ul li').length).to.eql(0)
+      notice = notification._prepare({timeout: "10000", content: "", title: ""})
+      expect($(notice).length).to.eql(1)
+      expect($(notice).attr('data-timeout')).to.not.eql("5000")
+      expect($(notice).attr('data-timeout')).to.eql("10000")
   
   describe '#clearAll', ->
     it "remove all four notifications", ->
-  
+      notification = new Notifications(getNotificationContent())  
+      expect(notification.el.find('ul li').length).to.eql(0)
+      notice = notification._prepare({type: "", content: "This is content", title: ""})
+      notification._postNotification(notice)
+      expect(notification.el.find('ul li').length).to.eql(1)
```

Comment:
There are a lot of repetitive assertions in here and you don't need many of them, especially those which test methods other than the one currently described, or test behaviour that's already tested elsewhere. For example, this spec has several assertions checking that `_postNotification(notice)` increases the list `length` by one, but that behaviour is already asserted by a different test, and anyway we're testing `clearAll()` here. So I think you can probably remove many of the `length` assertions through this file